cpu/stm32/include/gpio_ll_arch.h

```diff
@@ -77,7 +77,8 @@ static inline void gpio_ll_clear(gpio_port_t port, uword_t mask)
 #if defined(GPIO_BRR_BR0) && !defined(CPU_FAM_STM32F4)
     p->BRR = mask;
 #else
-    p->BSRR = mask << 16;
+    uint16_t *brr = (void *)&(p->BSRR);
```

Comment:
```suggestion
    volatile uint16_t *brr = (volatile uint16_t *)&(p->BSRR);
```

The `volatile` is needed here abd C++ is being annoying again here.